test/SparqlExpressionTest.cpp

```diff
@@ -947,6 +961,12 @@ TEST(SparqlExpression, DatatypeExpression) {
       IdOrLiteralOrIriVec{iriref("<http://example/iri/test#test>")});
   checkGetDatatype(IdOrLiteralOrIriVec{iriref("<http://example/iri/test>")},
                    IdOrLiteralOrIriVec{U});
+
+  // test corner case DatatypeValueGetter
+  TestContext ctx;
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      sparqlExpression::detail::DatatypeValueGetter{}(Id::max(), &ctx.context),
+      ::testing::ContainsRegex("should be unreachable"));
```

Comment:
Not a reason to stall this PR further (I will merge it tomorrow, as soon as the tests all have run through),
But for this simple non-regex case there also is `::testing::HasSubstring`.

Reply:
is noted